app/graphql/types/concerns/profile_scoring.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Types
+  # Methods related to profile scoring
+  module ProfileScoring
+    extend ActiveSupport::Concern
+
+    def compliant_host_count
+      ::CollectionLoader.for(
+        object.class,
+        object.policy_id ? :policy_test_result_hosts : :test_result_hosts
```

Comment:
Food for thought:
Where the result is dependent on a policy assignment, we could introduce a separate fields. For example `policyCompliantHostCount`. Each profile then can report it's own compliance and counts.